src/pybind/mgr/dashboard/frontend/src/app/ceph/smb/smb-domain-setting-modal/smb-domain-setting-modal.component.html

```diff
@@ -0,0 +1,121 @@
+<cds-modal size="md"
+           [open]="open"
+           [hasScrollingContent]="false"
+           (overlaySelected)="closeModal()">
+  <cds-modal-header (closeSelect)="closeModal()">
+    <h3 cdsModalHeaderHeading
+        i18n>{{ action | titlecase }} {{ resource | upperFirst }}</h3>
+  </cds-modal-header>
+
+  <div cdsModalContent>
+    <form name="domainSettingForm"
+          #formDir="ngForm"
+          [formGroup]="domainSettingsForm"
+          novalidate>
+      <div class="form-item">
+        <cds-text-label
+          labelInputID="realm"
+          [invalid]="
+            domainSettingsForm.controls.realm.invalid &&
+            (domainSettingsForm.controls.realm.touched || domainSettingsForm.controls.realm.dirty)
+          "
+          [invalidText]="realmNameError"
+          cdRequiredField="Realm Name"
+          i18n
+          >Realm Name
+          <input
+            cdsText
+            type="text"
+            placeholder="Realm name..."
+            id="realm"
+            name="realm"
+            formControlName="realm"
+            [invalid]="
+              domainSettingsForm.controls.realm.invalid &&
+              (domainSettingsForm.controls.realm.touched || domainSettingsForm.controls.realm.dirty)
+            "
+            [autofocus]="true"
+            modal-primary-focus
+          />
+        </cds-text-label>
+        <ng-template #realmNameError>
+          <span
+            class="invalid-feedback"
+            *ngIf="domainSettingsForm.showError('realm', formDir, 'required')"
+            i18n
+            >This field is required.</span
+          >
+        </ng-template>
+      </div>
+
+      <!-- Join Source -->
+      <ng-container
+        formArrayName="joinSources"
+        *ngFor="let joinSource of joinSources.controls; index as i"
+      >
+        <ng-container [formGroupName]="i">
+          <div cdsRow
+               class="form-item form-item-append">
+            <div cdsCol
+                 [columnNumbers]="{ lg: 7 }">
+              <input
+                cdsText
+                type="text"
+                placeholder="Ref.."
```

Comment:
I personally prefer `Id` over `Ref`
```suggestion
                placeholder="Id.."
```